Add buttons instead of dropdown in /design.

budzanowski opened the pull request:
Old:
![image](https://cloud.githubusercontent.com/assets/17271089/17963585/880afe42-6ab6-11e6-9470-84ed639e3b47.png)
New:
![image](https://cloud.githubusercontent.com/assets/17271089/17963665/f90d2264-6ab6-11e6-862e-67f57c382455.png)
This new look is hidden behind "manage/themes/magic_search" flag and visible only in dev.

I have chosen to create the change in form of a new module because amount of changes and necesity to hide the code behind feature flag was making the code unreadable. This way the implementation is clean and the flag is in only one place. This approach does not increase technical debt as the old module will be not needed and deleted when the new one is mature enough. This is also a good location to continue further work on Magic Search.


Test live: https://calypso.live/?branch=update/change-dropdown-to-buttons-in-design

klimeryk commented on config/development.json:
```diff
@@ -87,6 +87,7 @@
 		"manage/themes-jetpack": true,
 		"manage/themes/details": true,
 		"manage/themes/logged-out": true,
+		"manage/themes/magic_search": true,
```
The convention for features is to use `kebab-case` (yummy!), not :snake:_case.

seear commented:
it would be cool if _space_ or _return_ would select the button when it has keyboard focus:
![screen shot 2016-08-25 at 11 53 45](https://cloud.githubusercontent.com/assets/7767559/17966701/c1af813c-6aba-11e6-8875-4ebe283e0100.png)

(maybe not for this PR)

klimeryk commented on client/my-sites/themes/themes-magic-search-card/index.jsx:
```diff
@@ -0,0 +1,110 @@
+	},
+
+	render() {
+		const isJetpack = this.props.site && this.props.site.jetpack;
```
![19hagt](https://cloud.githubusercontent.com/assets/3392497/17966548/72456370-6ac2-11e6-893c-2a7f0701784c.jpg)

😜 
Multiple `const` declarations are OK, but if they are, for example, in different scopes. We discussed this on Slack and the current code guidelines have been [loosened a lot](https://github.com/Automattic/wp-calypso/commit/692d611e48a9c79ba33a8bc6a308db2b793d60f9) so it's not clear if this is allowed or not. IMHO, this convention looks a bit out of place in Calypso - @folletto can you weigh in here? 🙇

budzanowski replied:
If it will be the common consensus that we should do this way I don't mind and will follow the rest. On personal note I just don't feel that one type specifier for all declarations is good. I like when each declaration has type specified because I don't need to scroll(or at least limit the scroll) to see what type it is, and second is that I don't like the additional indentation required for one type specifier. That being said I know that the biggest value comes from consistency so maybe code guidelines should have more direct regulations here - so I will not be tempted to do it my way :)

folletto replied:
> It's preferable for `var`, `const`, and `let` statements to stay each one on an individual line

That's from the updated document. It's a "preference", so if one feels strongly to do otherwise, it's ok. My advice would be to try to have **one clear statement per line**, for a variety of reasons that however it's probably better to not discuss here. ;)

ehg replied:
> It's preferable for var, const, and let statements to stay each one on an individual line

YES! Way easier to refactor, and improves diff readability.

> it's probably better to not discuss here. ;)

Oops :)

seear commented:
Mobile view has the highlight chopped off on the right:
![screen shot 2016-08-25 at 12 03 23](https://cloud.githubusercontent.com/assets/7767559/17966895/104e25c2-6abc-11e6-9b9b-e77a1de42988.png)

Same thing happens in prod, but at least it is symmetrical there:
![screen shot 2016-08-25 at 12 03 38](https://cloud.githubusercontent.com/assets/7767559/17966911/2688d6d4-6abc-11e6-8743-e4c2fbf9e19e.png)

seear commented:
I'd say this can be merged pretty soon, since it only affects dev and we can iterate on it 👍

budzanowski commented:
Chopped-off highlight corrected.

folletto commented:
(1) Chopped right text in mobile size, no fade:

![screen shot 2016-08-26 at 11 12 05](https://cloud.githubusercontent.com/assets/4389/18002122/f61fe024-6b7d-11e6-8bf3-a0f8cddef6dd.png)

This is probably due to the hidden `×`, but we should fix it since it's a regression. We can probably do this in a separate PR however.

(2) Highlight padding: 

![screen shot 2016-08-26 at 11 13 39](https://cloud.githubusercontent.com/assets/4389/18002155/27eda82a-6b7e-11e6-8c4f-bad1422d3a76.png)

The margin you applied to fix the mobile cropping shows up badly in Desktop. I'd restore the previous condition, since the "proper" fix would be to make the entire box highlighted, so I'd rather keep the slight misalignment (or, if we can make it show up on mobile only) and work on the full highlight later.

Everything else seems working well! :)

budzanowski commented:
@folletto 
at (1) This PR uses old `<Search>` component so no fade and spacing left for icons - "new" `<Search>` addresses those issues #7485 - It would be already pulled if not the fact that it requires to check every place where `<Search>` is used especially places with custom CSS for `<Search>` 
at (2) I would not worry about it now - new `<Search>` is already made in a way that allows decoupling of highlight so we will have very simple code to add when we will want to highlight whole box.

folletto commented:
> I would not worry about it now - new <Search> is already made in a way that allows decoupling of highlight so we will have very simple code to add when we will want to highlight whole box.

Ok, let's just restore the styling that doesn't have the desktop glitch then for this PR. :)
